### PR TITLE
version-control-changdelog-actions

### PR DESCRIPTION
**Pull request by Mewken**
'Changesets' is installed to handle package versioning and changelogs

### CODE REVIEW

File: .github/workflows/changesets.yml [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+        uses: changesets/action@v1
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+          
+      - run: changeset version
+      - run: changeset release
+      
+    steps:
+      - git push: 
+      - git push: --tags
```
Comment (oskarhane) [inline .github/workflows/changesets.yml:36]: This should not be stand-alone steps, and the git stuff are commands.

```bash
- name: Git push changes back to origin
  run: git push && git push --tags
```

File: .github/workflows/changesets.yml [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+
+      - name: Create Release Pull Request
+        uses: changesets/action@v1
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+          
+      - run: changeset version
+      - run: changeset release
+      
+    steps:
```
Comment (oskarhane) [inline .github/workflows/changesets.yml:34]: Delete this row and move the step below up to the other steps, they're part of the same job

File: .github/workflows/changesets.yml [hunk 3]
```diff
@@ -0,0 +1,38 @@
+name: Changesets
+
+on:
+  push:
+    branches: [ main ]
```
Comment (oskarhane) [inline .github/workflows/changesets.yml:5]: It should not run on all commits but be triggered manually 

```suggestion
on: workflow_dispatch
```

File: .github/workflows/changesets.yml [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+      - name: Checkout Repo
+        uses: actions/checkout@v3
+
+      - name: Setup Node.js 16
+        uses: actions/setup-node@v3
+        with:
+          node-version: 16
+
+      - name: Install Dependencies
+        run: yarn
```
Comment (oskarhane) [inline .github/workflows/changesets.yml:22]: We don't use yarn in this project

File: .github/workflows/changesets.yml [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+
+      - name: Create Release Pull Request
+        uses: changesets/action@v1
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
```
Comment (oskarhane) [inline .github/workflows/changesets.yml:27]: I don't think we need this part

File: .github/workflows/changesets.yml [hunk 6]
```diff
@@ -0,0 +1,35 @@
...
+    runs-on: ubuntu-latest
+    
+    steps:
+      - name: Checkout Repo
+        uses: actions/checkout@v3
+
+      - name: Setup Node.js 16
+        uses: actions/setup-node@v3
+        with:
+          node-version: 16
```
Comment (oskarhane) [inline .github/workflows/changesets.yml:19]: We use node v18 in this project: https://github.com/neo4j/graph-schema-json-js-utils/blob/main/.nvmrc

File: .github/workflows/changesets.yml (hunk 4 above)
Comment (Mewken) [reply .github/workflows/changesets.yml:22]: Should I remove it completely or switch to npx or npm?

Comment (oskarhane) [reply .github/workflows/changesets.yml:22]: You should use `npm ci` here, to make sure that the lock-file isn't mutated during installation.

File: package.json [hunk 7]
```diff
@@ -18,5 +18,8 @@
   "homepage": "https://github.com/neo4j/graph-schema-json-js-utils#readme",
   "workspaces": [
     "packages/*"
-  ]
+  ],
+  "dependencies": {
+    "@changesets/cli": "^2.26.0"
```
Comment (oskarhane) [inline package.json:23]: This should be a "dev dependency", not a "dependency".

File: .github/workflows/changesets.yml [hunk 8]
```diff
@@ -0,0 +1,30 @@
+name: Changesets
```
Comment (oskarhane) [inline .github/workflows/changesets.yml:1]: Maybe change name to this workflow, to something like "Release" or similar, so it reflects what it does.

**Comment by Mewken** [comment]
Changesets is installed to handle package versioning and changelogs

**Comment by Mewken** [comment]
'Changesets' is installed to handle package versioning and changelogs

File: .github/workflows/changesets.yml [hunk 9]
```diff
@@ -0,0 +1,30 @@
...
+      - name: Checkout Repo
+        uses: actions/checkout@v3
+
+      - name: Setup Node.js 18
+        uses: actions/setup-node@v3
+        with:
+          node-version: 18
+
+      - name: Install Dependencies
+        run: npm
```
Comment (oskarhane) [inline .github/workflows/changesets.yml:22]:
```suggestion
        run: npm ci
```

File: .github/workflows/changesets.yml (hunk 8 above)
Comment (Mewken) [reply .github/workflows/changesets.yml:1]: ok

File: .github/workflows/changesets.yml (hunk 9 above)
Comment (Mewken) [reply .github/workflows/changesets.yml]: ok

File: package.json [hunk 10]
```diff
@@ -18,5 +18,8 @@
   "homepage": "https://github.com/neo4j/graph-schema-json-js-utils#readme",
   "workspaces": [
     "packages/*"
-  ]
+  ],
+  "devDependency": {
```
Comment (oskarhane) [inline package.json:22]: This is not correct: https://docs.npmjs.com/cli/v9/configuring-npm/package-json#devdependencies

**Review by oskarhane — APPROVED** [review]
LGTM 💯

**Pull request merged**